app/scripts/lib/require-on-demand.js

```diff
@@ -0,0 +1,82 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * Load modules asynchronously using requirejs. Useful to load modules
+ * that are not part of the main bundle. Calls to load a module return
+ * a promise that resolves once the module is loaded.
+ *
+ * Usage:
+ * requireOnDemand('passwordcheck')
+ *   .then(function (PasswordCheck) {
+ *      // use PasswordCheck here.
+ *   });
+ */
+
+define([
+  'lib/errors',
+  'lib/promise',
+  'underscore'
+], function (Errors, p, _) {
+  'use strict';
+
+  function requireOnDemand(resourceToGet) {
+    return p().then(function () {
+      var deferred = p.defer();
+
+      // requirejs takes care of ensuring only one outstanding request
+      // per module occurs if multiple requests are concurrently made
+      // for the same module.
+
+      // An alias to require is used to prevent almond from bundling
+      // the resource into the main bundle. Instead, the item will
+      // be loaded on demand, and the module returned when the promise
+      // resolves.
+      var getNow = window.require;
+      getNow(['nocache!' + resourceToGet],
+        deferred.resolve.bind(deferred), function (requireErr) {
+          // RequireJS errors described in
+          // http://requirejs.org/docs/api.html#errors
+          var errorType = requireErr.requireType || 'UNEXPECTED_ERROR';
```

Comment:
@vladikoff - we now handle requirejs errors, and [log them too](https://github.com/mozilla/fxa-content-server/pull/2983/files#diff-1c5e553272303abc152fd2aa8fd2b36aR46).